Source/WebCore/dom/ClassCollection.h

```diff
@@ -54,9 +56,14 @@ class ClassCollection final : public CachedHTMLCollection<ClassCollection, Colle
 
 inline ClassCollection::ClassCollection(ContainerNode& rootNode, CollectionType type, const AtomString& classNames)
     : CachedHTMLCollection(rootNode, type)
-    , m_classNames(classNames, rootNode.document().inQuirksMode() ? SpaceSplitString::ShouldFoldCase::Yes : SpaceSplitString::ShouldFoldCase::No)
     , m_originalClassNames(classNames)
 {
+    auto& document = rootNode.document();
+    bool shouldUseCaseInsensitive = document.decoder() && document.decoder()->encoding().name() == "UTF-8"_s
+        && document.doctype() && document.doctype()->publicId().isEmpty()
+        && document.doctype()->systemId().isEmpty();
```

Comment:
This expression should include the `inQuirksMode` check too instead of doing that separately.

A better way to check for UTF-8 encoding is to use the `UTF8Encoding()` function rather than a string constant containing `"UTF-8"`. We should write something more like this:

```suggestion
    auto isUTF8 = [](const Document& document) {
        return document.textEncoding() == PAL::UTF8Encoding();
    };
    auto hasEmptyIDs = [](const DocumentType* type) {
        return type && type->publicId().isEmpty() && type->systemId().isEmpty();
    };
    bool shouldFoldCase = document.inQuirksMode() || (isUTF8(document) && hasEmptyIDs(document.doctype());
```